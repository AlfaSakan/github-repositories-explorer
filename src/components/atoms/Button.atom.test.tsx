import { describe, it, expect, vi } from "vitest";
import { render, fireEvent, screen } from "@testing-library/react";
import Button from "./Button.atom";

describe("Button", () => {
  it("should match the snapshot", () => {
    const { container } = render(<Button loading={false} />);
    expect(container).toMatchSnapshot();
  });

  it("should match the snapshot when loading", () => {
    const { container } = render(<Button loading />);
    expect(container).toMatchSnapshot();
  });

  it("should invoke click function", async () => {
    const mockFn = vi.fn();
    render(<Button onClick={mockFn}>Button</Button>);
    fireEvent.click(screen.getByText("Button"));
    expect(mockFn).toHaveBeenCalled();
  });

  it("should not invoke click function", async () => {
    const mockFn = vi.fn();
    render(
      <Button disabled onClick={mockFn}>
        Button 1
      </Button>
    );
    fireEvent.click(screen.getByText("Button 1"));
    expect(mockFn).not.toHaveBeenCalled();
  });
});
