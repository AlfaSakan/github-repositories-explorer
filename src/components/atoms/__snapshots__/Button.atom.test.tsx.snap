// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should match the snapshot 1`] = `
<div>
  <button
    class="h-10 w-full bg-blue-500 text-white disabled:bg-neutral-400"
  />
</div>
`;

exports[`Button > should match the snapshot when loading 1`] = `
<div>
  <button
    class="h-10 w-full bg-blue-500 text-white disabled:bg-neutral-400"
  >
    <svg
      class="lucide lucide-loader-circle animate-spin inline"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 12a9 9 0 1 1-6.219-8.56"
      />
    </svg>
  </button>
</div>
`;
